enterprise/core-edge/src/main/java/org/neo4j/coreedge/server/core/CoreServerStartupProcess.java

```diff
@@ -85,10 +83,12 @@ public void start() throws Throwable
             {
                 caughtUp.get( joinCatchupTimeout, MILLISECONDS );
             }
-            catch ( InterruptedException | ExecutionException | TimeoutException e )
+            catch ( Throwable e )
             {
-                throw new RuntimeException( format( "Server failed to join cluster within catchup time limit [%d ms]",
-                        joinCatchupTimeout ), e );
+                String message =
+                        format( "Server failed to join cluster within catchup time limit [%d ms]", joinCatchupTimeout );
+                log.error( message, e );
```

Comment:
I don't know. `Lifecycle` is a mysterious beast I don't want to fight with for this improvement.